client/components/LoginPage.tsx

```diff
@@ -1,44 +1,93 @@
 import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { 
+  TextField, 
+  Box, 
+  Typography, 
+  InputAdornment,
+  IconButton,
+  Button
+} from "@mui/material";
+import {
+  Visibility,
+  VisibilityOff
+} from "@mui/icons-material";
+
 import { useLoginMutation } from "../features/api";
 
 const LoginPage = () => {
   const [login, { isLoading, isError, data }] = useLoginMutation();
 
+  const navigate = useNavigate();
+
+
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
```

Comment:
What does this do and why do we need it? Maybe add a comment to explain why.

Also might recommend putting this in-line with the onMouseDown prop below instead for readability.

Do you actually need to type the event? I thought that TS could infer, but maybe not

Reply:
TS didn't infer correctly when I was doing it. This is cut and pasted from the MUI docs. I think it's included so that the icon does not change with just a mouse down event, but rather a full click.

Reply:
I can put it inline in onMouseDown